src/CleanArchitecture.Mediator/Pipeline.cs

```diff
@@ -23,84 +24,84 @@ public class Pipeline : IPipeline
+        {
+            if (serviceFactory == null)
+                throw new ArgumentNullException(nameof(serviceFactory));
+
+            var _PipelineHandleAccessor = (IPipelineHandleAccessor)serviceFactory(typeof(PipelineHandleAccessor<>).MakeGenericType(this.GetType()));
+
+            this.m_PipelineHandle = _PipelineHandleAccessor.PipeHandle;
+        }
 
         #endregion Constructors
 
         #region - - - - - - Methods - - - - - -
 
-        private Invoker GetInvoker<TOutputPort>(IInputPort<TOutputPort> inputPort, TOutputPort outputPort)
-            => (Invoker)Activator.CreateInstance(
-                typeof(Invoker<,,>).MakeGenericType(this.GetType(), inputPort.GetType(), typeof(TOutputPort)),
-                inputPort,
-                outputPort);
-
         /// <summary>
         /// Invokes the pipeline.
         /// </summary>
         /// <typeparam name="TOutputPort">The type of output port.</typeparam>
         /// <param name="inputPort">The input to the pipeline.</param>
         /// <param name="outputPort">The output mechanism for the pipeline.</param>
+        /// <param name="serviceFactory">The factory used to get service instances.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be cancelled.</param>
         /// <exception cref="ArgumentNullException"><paramref name="inputPort"/> is null.</exception>
         /// <exception cref="ArgumentNullException"><paramref name="outputPort"/> is null.</exception>
         public Task InvokeAsync<TOutputPort>(
             IInputPort<TOutputPort> inputPort,
             TOutputPort outputPort,
+            ServiceFactory serviceFactory,
             CancellationToken cancellationToken)
             => inputPort == null
                 ? throw new ArgumentNullException(nameof(inputPort))
                 : outputPort == null
                     ? throw new ArgumentNullException(nameof(outputPort))
-                    : this.GetInvoker(inputPort, outputPort)
-                        .InvokePipelineAsync(this.m_ServiceFactory, cancellationToken);
+                    : PipelineInvoker.Instance(inputPort.GetType(), typeof(TOutputPort)).InvokePipelineAsync(inputPort, outputPort, this.m_PipelineHandle, serviceFactory, cancellationToken);
 
         #endregion Methods
 
         #region - - - - - - Nested Classes - - - - - -
 
-        private abstract class Invoker
-        {
-
-            #region - - - - - - Methods - - - - - -
-
-            public abstract Task InvokePipelineAsync(ServiceFactory serviceFactory, CancellationToken cancellationToken);
-
-            #endregion Methods
-
-        }
-
-        private class Invoker<TPipeline, TInputPort, TOutputPort> : Invoker
-            where TPipeline : IPipeline
-            where TInputPort : IInputPort<TOutputPort>
+        internal abstract class PipelineInvoker
```

Comment:
can these be private?

Reply:
Additionally, add a comment explaining that this class exists to change the reference type of the input port from IInputPort\<TOutputPort> to TInputPort where TInputPort : IInputPort\<TOutputPort>

Reply:
d